File: adventofcode-module/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day11/Part2.java
package nl.mvdr.adventofcode.adventofcode2024.day11;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Part2 extends PlutonianPebblesSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part2.class);

    public Part2() {
        super(75);
    }

    public static void main(String[] args) {
        var instance = new Part2();

        var result = instance.solve("input-day11-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day10/Part2.kt
package nl.mvdr.adventofcode.adventofcode2025.day10

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart2(lines: Sequence<String>) = lines.map(::parseMachine)
    .sumOf(Machine::computeFewestButtonPressesToRequiredJoltage)

fun main() {
    val result = FunctionSolver(::solvePart2).solve("input-day10.txt")
    logger.info { result }
}
